src/app/api/crud/base.py

```diff
@@ -41,16 +43,22 @@ async def fetch_all(
     query_filters: Optional[Dict[str, Any]] = None,
     exclusions: Optional[Dict[str, Any]] = None,
     limit: int = 50,
+    offset: Optional[int] = None,
+    query: Optional[Select] = None,
 ) -> List[Mapping[str, Any]]:
-    query = table.select().order_by(table.c.id.desc())
+    if query is None:
+        query = table.select()
     if isinstance(query_filters, dict):
         for key, value in query_filters.items():
             query = query.where(getattr(table.c, key) == value)
 
     if isinstance(exclusions, dict):
         for key, value in exclusions.items():
             query = query.where(getattr(table.c, key) != value)
-    return (await database.fetch_all(query=query.limit(limit)))[::-1]
+    query = query.order_by(table.c.id.desc()).limit(limit).offset(offset)
+    if isinstance(query, Query):
+        return [item.__dict__ for item in query[::-1]]
+    return (await database.fetch_all(query=query))[::-1]
```

Comment:
Good idea. Two questions though:
- I think I understand this is an equivalent of the offset syntax in SQL. But here we seem to be making an arbitrary choice on order direction. Should we make that a boolean param?
- regarding the query arg, we expect it to already have a `.select()`?

If we only want to paginate the queries, there are a few plugin options for this I think

Reply:
(paginate will prevent the case where we have a huge table and someone send `limit=100000` which might blow up the RAM)